climate-monitoring-client/src/main/java/climatemonitoring/Command.java

```diff
@@ -1,6 +1,64 @@
+/*
+
+Alessandro della Frattina 753073 VA
+Cristian Capiferri 752918 VA
+Francesco Lops 753175 VA
+Dariia Sniezhko 753057 VA
+
+*/
+
+
 package climatemonitoring;
 
+import java.util.StringTokenizer;
+
+
+/**
+ * The Command class processes a command line string to extract and store a command and its arguments, provides methods to retrieve these values, and includes constants for various predefined commands
+ * @author francescolops
+ * @version 1.0-SNAPSHOT
+ */
 class Command {
 
+	/**
+	 * Parses a command line string to extract a command and its associated arguments, storing them in m_cmd and m_args
+	 * @param line
+	 */
+	public void Command(String line){
+		
+		StringTokenizer st = new StringTokenizer(line);
+		if(st.hasMoreTokens()){
+
+			m_cmd = st.nextToken().substring(1);
+			m_args = st.nextToken("");
+		}
```

Comment:
When you check if the tokenizer has more tokens, you should check it for both m_cmd and m_args. You should also trim the strings. Example:
```
if (st.hasMoreTokens())
	m_cmd = st.nextToken().trim().substring(1);

if (st.hasMoreTokens())
	m_args = st.nextToken("").trim();
```